src/extensions/step_handlers/swupdate_handler/src/swupdate_handler.cpp

```diff
@@ -467,27 +492,46 @@ ADUC_Result SWUpdateHandlerImpl::IsInstalled(const tagADUC_WorkflowData* workflo
  */
 static ADUC_Result CancelApply(const char* logFolder)
 {
-    // Execute the install command with  "-r" to reverts the apply by
-    // telling the bootloader to boot into the current partition
-
-    // This is equivalent to : command << c_installScript << " -l " << logFolder << " -r"
-
-    std::vector<std::string> args{ adushconst::update_type_opt,       adushconst::update_type_microsoft_swupdate,
-                                   adushconst::update_action_opt,     adushconst::update_action_apply,
-                                   adushconst::target_log_folder_opt, logFolder };
-
     std::string output;
+    ADUC_Result result = { ADUC_Result_Failure_Cancelled, 0 };
 
-    const int exitCode = ADUC_LaunchChildProcess(ADUSHELL_FILE_PATH, args, output);
-    if (exitCode != 0)
+    const ADUC_ConfigInfo* config = ADUC_ConfigInfo_GetInstance();
+    if (config == nullptr)
     {
-        // If failed to cancel apply, apply should return SuccessRebootRequired.
-        Log_Error("Failed to cancel Apply, extendedResultCode = %d", exitCode);
-        return ADUC_Result{ ADUC_Result_Failure, exitCode };
+        Log_Error("Failed to get config info instance.");
+        result.ExtendedResultCode = ADUC_ERC_SWUPDATE_HANDLER_CANCEL_FAILED_TO_GET_CONFIG_INSTANCE;
+        goto done;
+    }
+
+    {
+        // Execute the install command with  "-r" to reverts the apply by
+        // telling the bootloader to boot into the current partition
+        // This is equivalent to : command << c_installScript << " -l " << logFolder << " -r"
+
+        const std::vector<std::string> aduShellArgs{
+            adushconst::config_folder_opt,     config->configFolder,
+            adushconst::update_type_opt,       adushconst::update_type_microsoft_swupdate,
+            adushconst::update_action_opt,     adushconst::update_action_apply,
+            adushconst::target_log_folder_opt, logFolder
+        };
+
+        const int exitCode = ADUC_LaunchChildProcess(config->aduShellFilePath, aduShellArgs, output);
+        if (exitCode != 0)
+        {
+            // If failed to cancel apply, apply should return SuccessRebootRequired.
+            Log_Error("Failed to cancel Apply, extendedResultCode = %d", exitCode);
+            result.ResultCode = ADUC_Result_Failure;
+            result.ExtendedResultCode = ADUC_ERC_SWUPDATE_HANDLER_CANCEL_FAILED_TO_CANCEL_APPLIED_UPDATE;
```

Comment:
Should it use MAKE_ERC... macro to pack the exitCode into lowest 20 bits into the corr facility + component?